src/rust/bitbox02-rust/src/workflow/transaction.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::bb02_async::option;
+use core::cell::RefCell;
+
+extern crate alloc;
+use alloc::boxed::Box;
+
+pub struct UserAbort;
+
+pub async fn verify_recipient(recipient: &str, amount: &str) -> Result<(), UserAbort> {
+    let result = RefCell::new(None as Option<Result<(), UserAbort>>);
+
+    let mut component = bitbox02::ui::confirm_transaction_address_create(
+        amount,
+        recipient,
+        Box::new(|ok| {
+            *result.borrow_mut() = Some(if ok { Ok(()) } else { Err(UserAbort) });
+        }),
+    );
+    component.screen_stack_push();
+    option(&result).await
+}
+
+pub async fn verify_total_fee(total: &str, fee: &str) -> Result<(), UserAbort> {
+    let result = RefCell::new(None as Option<Result<(), UserAbort>>);
+
+    let mut component = bitbox02::ui::confirm_transaction_fee_create(
+        total,
+        fee,
+        Box::new(|ok| {
+            *result.borrow_mut() = Some(if ok { Ok(()) } else { Err(UserAbort) });
+        }),
+    );
+    component.screen_stack_push();
```

Comment:
A random thought: a component is now popped from the screen stack on drop.
Here it's ok because we await the result but in general, it doesn't feel right to me because on drop of a specific component we just call `ui_screen_stack_pop()` which may result in popping a different one from what's dropped by rust.

Reply:
oversimplified example:

```rust
comp1.screen_stack_push();
comp2.screen_stack_push();
drop(comp1); // comp2 is actually popped
```

Reply:
of course it has nothing to do this PR in particular but should definitely do something about it.

Reply:
Yeah we were aware of this. Currently the workflows just have to be sure to push only once. Popping on drop is a requirement because drop is the mechanism to clean up, and the only way to clean up resources e.g. in an aborted async function call.

Maybe we can move the screen push into the component creation function in Rust, so at least push and pop are more close.

Fyi the same issue was in C. Not sure if this can be solved neatly without the component stack also ported to Rust.